src/runtime_src/core/tools/xbutil2/SubCmdAdvanced.cpp

```diff
@@ -0,0 +1,147 @@
+
+// 3rd Party Library - Include Files
+#include <boost/program_options.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <iostream> 
+
+#include "common/system.h"
+#include "common/device.h"
+#include <boost/format.hpp>
+#include <map>
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+SubCmdAdvanced::SubCmdAdvanced(bool _isHidden, bool _isDepricated, bool _isPreliminary)
+    : SubCmd("advanced", 
+             "Low level command operations")
+{
+  const std::string longDescription = "Low level command operations.";
+  setLongDescription(longDescription);
+  setExampleSyntax("");
+  setIsHidden(_isHidden);
+  setIsDeprecated(_isDepricated);
+  setIsPreliminary(_isPreliminary);
+}
+
+
+void
+SubCmdAdvanced::execute(const SubCmdOptions& _options) const
+{
+  XBU::verbose("SubCommand: advanced");
+
+  // ============= Define all of the options =================================
+  po::options_description allDesc;
+
+  // -- Common top level options ---
+  bool help = false;
+
+  po::options_description commonTopDesc("Common Options"); 
+  commonTopDesc.add_options()
+    ("help,h", boost::program_options::bool_switch(&help), "Help to use this sub-command")
+  ;
+  allDesc.add(commonTopDesc);
+
+  // -- Define the supporting option options ----
+  SubOptionOptions subOptionOptions;
+  subOptionOptions.emplace_back(std::make_shared<OO_Clock>("clock"));
+  subOptionOptions.emplace_back(std::make_shared<OO_MemRead>("read-mem"));
+  subOptionOptions.emplace_back(std::make_shared<OO_MemWrite>("write-mem"));
+  subOptionOptions.emplace_back(std::make_shared<OO_P2P>("p2p"));
+
+  for (auto subOO : subOptionOptions) {
+    allDesc.add_options()(subOO->longName().c_str(), subOO->description().c_str());
+    subOO->setExecutable(getExecutableName());
+    subOO->setCommand(getName());
+  }
+
+  // =========== Process the options ========================================
+
+  // 1) Process the common top level options 
+  po::parsed_options parsedCommonTop = 
+    po::command_line_parser(_options).
+    options(allDesc).          
+    allow_unregistered().           // Allow for unregistered options
+    run();                          // Parse the options
+
+  po::variables_map vm;
+
+  try {
+    po::store(parsedCommonTop, vm);  // Can throw
+    po::notify(vm);                  // Can throw (but really isn't used)
+
+    // Multual DRC
+    for (unsigned index1 = 0; index1 < subOptionOptions.size(); ++index1) {
+      for (unsigned index2 = index1 + 1; index2 < subOptionOptions.size(); ++index2) {
+        conflictingOptions(vm, subOptionOptions[index1]->longName(), subOptionOptions[index2]->longName());
+      }
+    }
+  } catch (const std::exception & e) {
+    std::cerr << "ERROR: " << e.what() << std::endl;
+    printHelp(allDesc, subOptionOptions);
+    return;
+  }
+
+  // Find the subOption;
+  std::shared_ptr<OptionOptions> optionOption;
+  for (auto subOO : subOptionOptions) {
```

Comment:
& is important otherwise you incr/decr refcount

Reply:
Done